server/controllers/finance/reports/reportAccounts/index.js

```diff
@@ -40,8 +45,15 @@ function document(req, res, next) {
     return next(e);
   }
 
-  return queryReportAccount(params.account_id, params.sourceId)
+  return queryReportAccount(params.account_id, params.sourceId, params.dateFrom, params.dateTo)
     .then((result) => {
+
+      result.accounts.forEach(function (account) {
+        cumulBalance += account.debit - account.credit;
```

Comment:
As much as possible all calculations for balances/ debits and credits should be done in MySQL and not in Javascript. As referenced in other issues/ pull requests Javascript has many issues when it comes to its Maths library that could cause issues in the long run. 

In this case I believe you can calculate a running total in MySQL by using a variable. Something like: 

```sql 
SET @cumulative_balance := 0;
SELECT 
debit, 
credit, 
(@cumulative_balance := @cumulative_balance + (debit - credit)) as cumulative_balance
FROM #...
```